### PR TITLE
[catalog] Impl flink catalog use mysql

### PR DESCRIPTION
**Pull request by GOODBOY008**
WIP......

### CODE REVIEW

File: streampark-flink/streampark-flink-catalog-mysql/src/main/java/org/apache/streampark/flink/catalog/StreamParkMysqlCatalog.java [hunk 1]
```diff
@@ -0,0 +1,985 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+@Getter
+public class StreamParkMysqlCatalog extends AbstractCatalog {
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-catalog-mysql/src/main/java/org/apache/streampark/flink/catalog/StreamParkMysqlCatalog.java]: I don't like the name `StreamParkMysqlCatalog`, what about `MysqlCatalog`

**Comment by GOODBOY008** [comment]
@wolfboys I had remove dependency `streampark-common` , beacuse of `Tuple` class is exist in minority dependencies.

File: streampark-flink/streampark-flink-catalog-mysql/src/main/java/org/apache/streampark/flink/catalog/MysqlCatalog.java [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+      ObjectPath tablePath,
+      CatalogPartitionSpec partitionSpec,
+      CatalogColumnStatistics columnStatistics,
+      boolean ignoreIfNotExists)
+      throws CatalogException {
+    throw new UnsupportedOperationException();
+  }
+
+  private boolean mapNotEmpty(Map<?, ?> map) {
+    return map == null || map.isEmpty();
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-catalog-mysql/src/main/java/org/apache/streampark/flink/catalog/MysqlCatalog.java]: map != null && !map.isEmpty()

File: streampark-flink/streampark-flink-catalog-mysql/src/main/java/org/apache/streampark/flink/catalog/dao/BaseDao.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  public void jdbcUpdate(String sql, Object... params) throws SQLException {
+    new QueryRunner(dataSource).update(sql, params);
+  }
+
+  public <T> T jdbcInsert(String sql, ResultSetHandler<T> rsh, Object... params)
+      throws SQLException {
+    return new QueryRunner(dataSource).insert(sql, rsh, params);
+  }
+
+  public void jdbcBatch(String sql, Object[][] params) throws SQLException {
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-catalog-mysql/src/main/java/org/apache/streampark/flink/catalog/dao/BaseDao.java]: unused method. we can delete the method

File: streampark-flink/streampark-flink-catalog-mysql/pom.xml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    <properties>
+        <flink.version>1.16.2</flink.version>
+        <HikariCP.version>4.0.3</HikariCP.version>
+        <mysql-connector-j.version>8.0.33</mysql-connector-j.version>
+    </properties>
+
+    <dependencies>
+
+        <dependency>
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-catalog-mysql/pom.xml:40]: We need to statement the license of these dependencies to [LICENSE](https://github.com/apache/incubator-streampark/blob/dev/dist-material/release-docs/LICENSE) file

**Pull request closed without merging**